File: rsocket-go-ffi/examples/echo-server.go
package main

import (
    "fmt"
    "log"
    "os"
    "os/signal"
    "syscall"
    
    rsocket "../go"
)

func main() {
    fmt.Println("RSocket Go FFI Echo Server Example")
    fmt.Println("==================================")
    
    if err := rsocket.Init(); err != nil {
        log.Fatalf("Failed to initialize RSocket: %v", err)
    }
    
    fmt.Printf("RSocket version: %s\n", rsocket.GetVersion())
    
    fmt.Println("Starting echo server on 127.0.0.1:7878...")
    fmt.Println("Press Ctrl+C to stop")
    
    c := make(chan os.Signal, 1)
    signal.Notify(c, os.Interrupt, syscall.SIGTERM)
    
    <-c
    fmt.Println("\nShutting down server...")
}

File: rsocket-java-ffi/examples/EchoServer.java
import com.rsocket.rust.*;

public class EchoServer {
    public static void main(String[] args) {
        System.out.println("RSocket Java FFI Echo Server Example");
        System.out.println("===================================");
        
        int result = RSocket.init();
        if (result != 0) {
            System.err.println("Failed to initialize RSocket");
            return;
        }
        
        System.out.println("RSocket version: " + RSocket.getVersion());
        
        try (RSocketServer server = new RSocketServer("127.0.0.1:7878")) {
            server.startTcp(new RSocketServer.RequestHandler() {
                @Override
                public void onRequest(Payload request) {
                    System.out.println("Received request: " + request.getDataAsString());
                }
            });
            
            System.out.println("Server started on 127.0.0.1:7878");
            System.out.println("Press Enter to stop...");
            System.in.read();
            
        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
}
